File: database/entities/post.ts
import { BaseEntity } from './base';
import { Column, Entity, JoinTable, ManyToMany } from 'typeorm';
import { Category } from './category';

@Entity({ name: 'posts' })
export class Post extends BaseEntity {
  @Column({
    type: 'character varying',
    nullable: false,
  })
  title: string;

  @Column({
    type: 'character varying',
    nullable: true,
  })
  description: string;

  @Column({
    type: 'character varying',
    nullable: true,
  })
  image: string;

  @ManyToMany(() => Category, (category) => category.posts, {
    cascade: true,
  })
  @JoinTable()
  categories: Category[];
}

File: server/database/models/post.py
from typing import List

from sqlalchemy import Column, String
from sqlalchemy.orm import relationship, Mapped

from .base_model import BaseModel
from .category import Category
from ..relations.posts_categories import posts_categories


class Post(BaseModel):
    __tablename__ = 'posts'

    title: str = Column(String, nullable=False)
    description: str = Column(String)
    image: str = Column(String)
    categories: Mapped[List[Category]] = relationship(secondary=posts_categories)
